pkg/cmd/roachtest/tests/restore.go

```diff
@@ -430,7 +430,7 @@ func (hw hardwareSpecs) makeClusterSpecs(r registry.Registry) spec.ClusterSpec {
 	}
 	s := r.MakeClusterSpec(hw.nodes, clusterOpts...)
 
-	if s.Cloud == "aws" && s.VolumeSize != 0 {
+	if backupCloud == spec.AWS && s.Cloud == spec.AWS && s.VolumeSize != 0 {
```

Comment:
The second conjunct is a tautology because `s.Cloud` is empty in this case. Incidentally, none of the existing roachtests set `s.Cloud` to anything other than `gce`. 🤷

Reply:
@srosenberg I don't believe that's the case, at least on master anymore. In an ad hoc test, I added a print statement in this branch to confirm we take it. We've also confirmed in later restore roachtest failures, that we properly removed the `d` in the aws machine type specification. (s[ee here](https://github.com/cockroachdb/cockroach/issues/100804#issuecomment-1501746465)). 

Also, if s.Cloud were not set to `aws`, I don't think any aws roachtests would work? We seem to [rely](https://github.com/cockroachdb/cockroach/blob/e640de72f80af91beeff2256e8fb25c966c2b845/pkg/cmd/roachtest/spec/cluster_spec.go#L234) on this variable to set the right machine type.

Reply:
Nm, you're right. I missed the fact that `Cloud` is passed (`r.Cloud` below) from the `testRegistryImply`, which in turns gets initialized before tests are registered,


```
spec.MakeClusterSpec(r.cloud, r.instanceType, nodeCount, finalOpts...)
```